woocommerce/woocommerce-ios
# Variations: Add error state for syncing global attribute options

Closes https://github.com/woocommerce/woocommerce-ios/issues/3601.

## Description

This PR adds error state on attribute options screen.

## Changes

1. Switch `isSyncing` from `Bool` to `SyncState` enum.
2. Add `.failed` case for `SyncState`
3. Display `EmptyStateViewController` with error description and retry action for `.failed` case in `AddAttributeOptionsViewController`.

`EmptyStateViewController` picked instead of non-blocking `Notice` because we need to block user from adding new options until existing data is properly synced.

## Test

1. Open variable product, with existing variations or not
2. Tap "variations" (or "add variation")
3. For no variations:
a. Tap "Add Variation"
For existing variations:
a. Tap "•••" in navbar, pick "Edit Attributes"
b. Tap "Add New Attribute"

4. Turn off internet connection
5. Select global attribute
6. Observe error state
7. Turn on internet connection and tap "Retry"
8. Observe correctly synced options list

## Screenshots

before | after
--|--
![Simulator Screen Shot - iPhone 12 - 2021-02-18 at 12 37 39](https://user-images.githubusercontent.com/3132438/108337425-27423800-71e6-11eb-8727-c9f81ce43e0e.png)|![Simulator Screen Shot - iPhone 12 - 2021-02-18 at 12 38 35](https://user-images.githubusercontent.com/3132438/108337526-3fb25280-71e6-11eb-8389-3680a1ed746a.png)


Update release notes:
